prov/net/src/xnet_eq.c

```diff
@@ -118,6 +124,52 @@ static struct fi_ops xnet_eq_fi_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
+static int xnet_eq_wait_try_func(void *arg)
+{
+	(void)arg;
+	return FI_SUCCESS;
+}
+
+int xnet_eq_add_progress(struct xnet_eq *eq, struct xnet_progress *progress,
+			 void *context)
+{
+	return ofi_wait_add_fd(eq->util_eq.wait,
+			       ofi_dynpoll_get_fd(&progress->epoll_fd),
+			       POLLIN, xnet_eq_wait_try_func, NULL, context);
+}
+
+int xnet_eq_del_progress(struct xnet_eq *eq, struct xnet_progress *progress)
+{
+	return ofi_wait_del_fd(eq->util_eq.wait,
+			       ofi_dynpoll_get_fd(&progress->epoll_fd));
+}
+
+static int xnet_eq_monitor_all(struct xnet_eq *eq, struct xnet_fabric *fabric)
+{
+	struct xnet_domain *domain;
+	struct dlist_entry *item;
+	int ret;
+
+	ret = xnet_eq_add_progress(eq, &fabric->progress,
+				   &fabric->util_fabric.fabric_fid);
+	if (ret)
+	    return ret;
+
+	ofi_mutex_lock(&fabric->util_fabric.lock);
+	dlist_foreach(&fabric->util_fabric.domain_list, item) {
+		domain = container_of(item, struct xnet_domain,
+				      util_domain.list_entry);
+		ret = xnet_eq_add_progress(eq, &domain->progress,
+					   &domain->util_domain.domain_fid);
+		if (ret)
+		    break;
+	}
+	if (ret == 0)
+	    dlist_insert_tail(&eq->wait_eq_entry, &fabric->wait_eq_list);
```

Comment:
Using spaces instead of tab.

Reply:
A failure is unlikely, but the cleanup here isn't complete.  The wait set could have been created by the user.  We may have added some progress epoll_fd's to the wait epoll fd.  We need to remove any that we added.

We need to do the same thing when destroying this EQ.  The destroy case is a little easier, as we can basically just reverse this call.  Cleanup in the error case here is a pain, but we can walk the list backwards from where an error occurred (plus release the fabric).

Reply:
I reworked the error handling according to your review comment. 
Please let me know if it's OK for you now.